python/GafferUI/ContextQueryUI.py

```diff
@@ -0,0 +1,408 @@
+			if the variable does not exist ( or is incompatible ).
+			""",
+
+		],
+
+		"out.*.value..." : [
+
+			"noduleLayout:label", functools.partial( __getLabel, parentPlug = "values" ),
+
+		],
+
+	}
+)
+
+##########################################################################
+# _ContextQueryFooter
+##########################################################################
+
+class _ContextQueryFooter( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, plug ) :
+
+		row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal )
+
+		GafferUI.PlugValueWidget.__init__( self, row, plug )
+
+		with row :
+
+			GafferUI.Spacer( imath.V2i( GafferUI.PlugWidget.labelWidth(), 1 ) )
+
+			GafferUI.MenuButton(
+				image = "plus.png",
+				hasFrame = False,
+				menu = GafferUI.Menu( Gaffer.WeakMethod( self.__menuDefinition ) )
+			)
+
+			GafferUI.Spacer( imath.V2i( 1 ), imath.V2i( 999999, 1 ), parenting = { "expand": True } )
+
+		plug.node().plugSetSignal().connect(
+			Gaffer.WeakMethod( self.__updateQueryMetadata ),
+			scoped = False
+		)
+
+	def _updateFromPlug( self ) :
+
+		self.setEnabled( self._editable() )
+
+	def __menuDefinition( self ) :
+
+		result = IECore.MenuDefinition()
+
+		for item in [
+			Gaffer.BoolPlug,
+			Gaffer.FloatPlug,
+			Gaffer.IntPlug,
+			"NumericDivider",
+			Gaffer.StringPlug,
+			"StringDivider",
+			Gaffer.V2iPlug,
+			Gaffer.V3iPlug,
+			Gaffer.V2fPlug,
+			Gaffer.V3fPlug,
+			"VectorDivider",
+			Gaffer.Color3fPlug,
+			Gaffer.Color4fPlug,
+		] :
+			if isinstance( item, six.string_types ) :
+				result.append( "/" + item, { "divider": True } )
+			else :
+				result.append(
+					"/" + item.__name__.replace( "Plug", "" ),
+					{
+						"command" : functools.partial( Gaffer.WeakMethod( self.__addQuery ), "", item ),
+					}
+				)
+
+		return result
+
+	def __addQuery( self, name, plugTypeOrValue ) :
+
+		with Gaffer.UndoScope( self.getPlug().ancestor( Gaffer.ScriptNode ) ) :
+
+			node = self.getPlug().node()
+
+			if isinstance( plugTypeOrValue, IECore.Data ) :
+				dummyPlug = Gaffer.PlugAlgo.createPlugFromData(
+					"dummyPlug",
+					Gaffer.Plug.Direction.In,
+					Gaffer.Plug.Flags.Default,
+					plugTypeOrValue
+				)
+				node.addQuery( dummyPlug, name)
+			else:
+				node.addQuery( plugTypeOrValue(), name)
+
+	def __updateQueryMetadata( self, plug ) :
+
+		node = plug.node()
+
+		if node["queries"].isAncestorOf( plug ) :
+
+			qPlug = plug.ancestor( Gaffer.NameValuePlug )
+
+			if qPlug is not None and qPlug["name"] == plug :
+
+				Gaffer.Metadata.plugValueChangedSignal( node )(
+					node.outPlugFromQueryPlug( qPlug ),
+					"label",
+					Gaffer.Metadata.ValueChangedReason.StaticRegistration
+				)
+
+##########################################################################
+# Delete Plug
+##########################################################################
+
+
+def __plugPopupMenu( menuDefinition, plugValueWidget ) :
```

Comment:
We discussed having a `Create Context Query...` menu item on plug popup menus, in the same vein as `Create Python Expression...`. I think it's worth doing that now, as it'll make it a lot easier for the less technical user to wire one up.